### DISCUSSION TITLE
Two questions about the descriptor se_a_mask and the descriptor se_e2_a.

### DISCUSSION THREAD

**Discussion by ZhouXY-PKU** [deepmodeling/deepmd-kit#3868] [Q&A]
Hello everyone, 

I have the following two  questions about the descriptor [`se_a_mask`](https://docs.deepmodeling.com/projects/deepmd/en/r2/model/train-se-a-mask.html) and the descriptor [`se_e2_a`](https://docs.deepmodeling.com/projects/deepmd/en/r2/model/train-se-e2-a.html).

The first is that, if we use  `se_a_mask`  for a PBC system (I know DP not support that yet, just an assumption), should the results almost be the same with that if we use  `se_e2_a`, or the accuracy of the two DP models can be significantly different (because the descriptors are not the same one)?

The second is that, as `se_e2_a` has no padding, should the numbers of atoms in all data frames for _one mini-batch gradient descent step_ in training must be same , or can we use data frames with different numbers of atoms for _one step of  mini-batch gradient descent_? (when we use `se_e2_a` to train, the data frames with different numbers of atoms are used in _different step of  mini-batch gradient descent_, is that right? )

Thank you very much!

**Comment by njzjz** [comment]
FYI @dingye18 for `se_a_mask`

`se_e2_a` requires the same number of atoms, but `se_atten_v2` does not. You can set [`batch_size`](https://docs.deepmodeling.com/projects/deepmd/en/r2/train/train-input.html#argument:training/training_data/batch_size) to `mixed:N` with `se_atten_v2`.

**Reply by ZhouXY-PKU** [reply]
Thank you very much!
